### PR TITLE
Added soft memory limit even when hard is selected

### PR DESCRIPTION
**Pull request by lukaszmatczak**
When hard memory limit is set, OOM is called even if buffers/cache can be released to free memory. To avoid this, soft limit can be enabled in addition to hard. I propose to set soft limit at 90% of hard limit value. It's mentioned in cgroup documentation:
https://www.kernel.org/doc/Documentation/cgroup-v1/memory.txt
> NOTE2: It is recommended to set the soft limit always below the hard limit, otherwise the hard limit will take precedence. 

Signed-off-by: Lukasz Matczak <lukasz.matczak@cba.pl>

### CODE REVIEW

**Comment by stgraber** [comment]
That sounds good but you'll also need to set the soft limit in initLXC() so that it applies at container startup too, not just when changed after the container is running.

**Comment by stgraber** [comment]
Marking as incomplete.

**Comment by lukaszmatczak** [comment]
Added. Is it complete now?

**Comment by stgraber** [comment]
jenkins: test this please

File: lxd/container_lxc.go [hunk 1]
```diff
@@ -3502,6 +3507,12 @@ func (c *containerLXC) Update(args containerArgs, userRequested bool) error {
 							return err
 						}
 					}
+					// Set soft limit to value 10% less than hard limit
+					err = c.CGroupSet("memory.soft_limit_in_bytes", memory*0.9)
```
Comment (stgraber) [inline lxd/container_lxc.go]: memory here is of type string, not an integer so you can't do *0.9

**Comment by stgraber** [comment]
Pushed a commit to fix the Jenkins failure.

**Comment by stgraber** [comment]
jenkins: test this please

**Comment by stgraber** [comment]
jenkins: test this please

**Comment by stgraber** [comment]
@brauner should be good to merge, buildd04 rebooted halfway through tests.

**Pull request merged**